pkg/k8s/ingresses/translate.go

```diff
@@ -0,0 +1,132 @@
+package ingresses
+
+import (
+	"github.com/okteto/okteto/pkg/model"
+	networkingv1 "k8s.io/api/networking/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Translate Endpoint to Ingress
+// Translate Service to Ingress
+
+type TranslateOptions struct {
+	Name      string
+	Namespace string
+}
+
+// TranslateEndpoint translates the endpoints spec at compose or okteto manifest and returns an ingress
+func TranslateEndpoint(endpointName string, endpoint model.Endpoint, opts *TranslateOptions) *Ingress {
+	if endpointName == "" {
+		endpointName = opts.Name
+	}
+	return &Ingress{
+		V1:      translateEndpointIngressV1(endpointName, endpoint, opts),
+		V1Beta1: translateEndpointIngressV1Beta1(endpointName, endpoint, opts),
+	}
+}
+
+func translateEndpointIngressV1(ingressName string, endpoint model.Endpoint, opts *TranslateOptions) *networkingv1.Ingress {
```

Comment:
Minor comment: same about the naming, I think we can remove endpoint from the name